### DISCUSSION TITLE
egui_extras::Table: Detecting clicks on cells/rows

### DISCUSSION THREAD

**Discussion by crumblingstatue** [emilk/egui#1519] [Q&A]
I am making an egui frontend to a package manager.
It's using the Table widget from egui_extras.

![image](https://user-images.githubusercontent.com/1521976/164292978-98d09dc6-6477-4ba5-ab2c-5b8e2c31c8f9.png)

Right now I'm using buttons in the name cell to allow clicking it to get more information about a package.
Is there a way to detect a click on a cell, or even a whole row, so I can just use a normal label, and the entire cell/row is clickable?

**Comment by emilk** [comment]
Not right now, but it would be a good thing to add. I think it is important that if we add it, the row also responds properly to hover (i.e. has a hover effect, just like a button).

**Comment by max-huster** [comment]
It seems that the function that allocates the UI Area for the Columns does not enable click sensing (for a good reason)

I branched and changed this: https://github.com/emilk/egui/compare/master...max-huster:egui:master#diff-edeef80b2fe77a89f96f0385200cc60a05a2df8d01b3eb8301038687a5c4b501


```
// in your Table Builder in the .body Method add:

let mut clicked = false;
// Keep track whether a column is clicked
clicked |= row.col(|ui| {
    ui.label("First Column");
}).1.clicked();

clicked |= row.col(|ui| {
    ui.label("Second Column");
}).1.clicked();

if(clicked){
    log::info!("Row Clicked!!!");
}
```

For my project that works. But when you add a button in the column, the sensing there does not work anymore. So maybe add a seperate method like `.clickable_col()` or `.interactive_col()`.

**Comment by max-huster** [comment]
In [my fork](https://github.com/max-huster/egui) i added `col_sense` function which solves the formerly described problem. There is a seperate function in the `TableRow` called `col_sense` on which you can specify the Sense. In my case `Sense::click()`. The "normal" `.col` method behaves exactly the same as before.

Here is the example code:
```
let mut clicked = false;
// Keep track whether a column is clicked
clicked |= row.col_sense(Sense::click(), |ui| {
    ui.label("First Column");
}).1.clicked();

clicked |= row.col_sense(Sense::click(), |ui| {
    ui.label("Second Column");
}).1.clicked();

row.col(|ui|{
    if(ui.button("TEST").clicked()){
        log::info!("Button Click still works!");
    }
});


if(clicked){
    log::info!("Row Clicked!!!");
}
```
@emilk should i open a pull request?

**Reply by Tudyx** [reply]
Would be great! I will be happy to help if you need any

**Reply by max-huster** [reply]
Just opened https://github.com/emilk/egui/pull/2904
